app/models/transaction.rb

```diff
@@ -158,6 +158,11 @@ def get_type
     end
   end
 
+  def viewed_for_member?(member)
+    return false if member&.last_visited_at.blank?
```

Comment:
@yunixon return `true` if `last_visited_at` blank. For first visit it's better to show all transactions as viewed.